src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -577,12 +587,17 @@ void GcsActorManager::OnWorkerDead(const ray::ClientID &node_id,
     actor_id = gcs_actor_scheduler_->CancelOnWorker(node_id, worker_id);
   }
 
-  if (!actor_id.IsNil()) {
-    RAY_LOG(WARNING) << "Worker " << worker_id << " on node " << node_id
-                     << " failed, restarting actor " << actor_id;
-    // Reconstruct the actor.
-    ReconstructActor(actor_id, /*need_reschedule=*/!intentional_exit);
+  // If actor is not created & not in the creation process, don't do anything.
+  if (actor_id.IsNil()) {
```

Comment:
move it to line 588?

Reply:
This is equivalent to the previous logic. Do you mean to change it like before?

Reply:
I'm not sure if the actor_id returned on line 581 will be nil.

Reply:
Will move it inside the else statement!